### DISCUSSION TITLE
Decide whether or not to install and deps and if yes choose from package managers

### DISCUSSION THREAD

**Discussion by kareemmahlees** [vercel/next.js#57969] [Ideas]
### Goals

1. Ask the user whether or not he wants to install the deps
2. If yes, he would then be prompted to choose between different package managers
3. Perhaps add an additional option at the end of the package managers list that says "Something Else" if the user wants to use for example bun. 

### Non-Goals

1.
2.
3.

### Background

Currently, the workaround here is to delete the node_modules folder and then reinstall the deps with another package manager.
This is of course a resource-consuming method.

### Proposal

The list of package managers should only be prompted if the user has chosen to install the deps.
Upon choosing, the `install` command of the corresponding package manager should be run.

I would be happy to contribute in making this possible 🙂

**Comment by icyJoseph** [comment]
Maybe I am wrong, but if you run either of:

```console
npx create-next-app@latest
# or
yarn create next-app
# or
pnpm create next-app
# or
bunx create-next-app
```

Then, it'll use the package manager you ran the command with, right?

**Reply by kareemmahlees** [reply]
@icyJoseph That's correct, but it is not documented that the command is runnable with other package managers, so we can fix that!

**Reply by icyJoseph** [reply]
- https://www.npmjs.com/package/create-next-app

<img width="867" alt="Screenshot 2023-11-03 at 09 42 04" src="https://github.com/vercel/next.js/assets/21013447/4088af4b-687a-4c09-bfca-ecda94ff3075">

**Reply by kareemmahlees** [reply]
@icyJoseph is it on the website docs also ?

**Reply by icyJoseph** [reply]
Yes it is:

- https://nextjs.org/docs/app/api-reference/create-next-app

<img width="1344" alt="Screenshot 2023-11-03 at 09 45 00" src="https://github.com/vercel/next.js/assets/21013447/8ae66580-c2d5-42a2-8bc9-e18f69adcba2">
